### PR TITLE
Created basic structure

### PR DESCRIPTION
**Pull request by serhii-shyian**
1) Add checkstyle.xml in the root, pointed in pom.xml and build part
2) Add dependencies in pom.xml
3) Add application.properties in resources package
4) Add init_db.sql in resources package
5) Add application.properties in test/resources package
6) Add Book class in model package
7) Add DataProcessingException in exception package
8) Add BookRepository in repository package and BookRepositoryImpl in repository/impl package
9) Add BookService in service package and BookServiceImpl in service/impl package

### CODE REVIEW

File: src/main/java/com/example/bookstore/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+@Entity
+@Table(name = "books")
+@Data
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NonNull
+    private String title;
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/model/Book.java:21]: use @ column nullable = false

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookstore/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Table(name = "books")
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/model/Book.java:14]: Don't use `@Data` for entities, find article about it

File: src/main/java/com/example/bookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    private String author;
+    @Column(nullable = false, unique = true)
+    private String isbn;
+    @Column(nullable = false)
+    private BigDecimal price;
+    private String description;
+    @Column(name = "cover_image")
+    private String coverImage;
+
+    public Book() {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/model/Book.java:31]: You don't need to create default constructor for entity if you don't have any with parameters

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Provide Main class with implementation according to the task: Class with @SpringBootApplication annotation that will have a CommandLineRunner bean.

File: src/main/java/com/example/bookstore/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Table(name = "books")
+@Getter
+@Setter
+@ToString
+@NoArgsConstructor
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/model/Book.java:20]: In Java constructor without parameters is created by default when you create an object: Book book = new Book(); 
If you have at least one constructor with parameters, you must create default constructor by yourself for Entity.

Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/model/Book.java:20]:
```suggestion
```

File: src/main/java/com/example/bookstore/model/Book.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    @Column(nullable = false)
+    private String title;
+    @Column(nullable = false)
+    private String author;
+    @Column(nullable = false, unique = true)
+    private String isbn;
+    @Column(nullable = false)
+    private BigDecimal price;
+    private String description;
+    @Column(name = "cover_image")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/model/Book.java:34]: Hibernate autogenerate the name using snake_case
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's make it perfect!

**Review by Elena-Bruyako — APPROVED** [review]
Well done!

**Pull request merged**